File: client/src/app/entities/vehicle.ts
import {User} from './user';
import {Vehicletype} from './vehicletype';
import {DataPage} from '../shared/data-page';
import {Vehiclestatus} from './vehiclestatus';

export class Vehicle {
  id: number;
  no: string;
  modal: string;
  brand: string;
  payloadwidth: number;
  payloadlength: number;
  payloadheight: number;
  vehicletype: Vehicletype;
  vehiclestatus: Vehiclestatus;
  photo: string;
  description: string;
  creator: User;
  tocreation: string;
}

export class VehicleDataPage extends DataPage{
    content: Vehicle[];
}

File: server/src/main/java/bit/project/server/entity/Vehicle.java
package bit.project.server.entity;

import lombok.Data;
import java.util.List;
import javax.persistence.*;
import javax.persistence.Id;
import javax.persistence.Lob;
import java.time.LocalDateTime;
import lombok.NoArgsConstructor;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Vehicle{
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String no;

    private String modal;

    private String brand;

    private Integer payloadwidth;

    private Integer payloadlength;

    private Integer payloadheight;

    private String photo;

    @Lob
    private String description;

    private LocalDateTime tocreation;


    @ManyToOne
    private Vehicletype vehicletype;

    @ManyToOne
    private Vehiclestatus vehiclestatus;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;


    @JsonIgnore
    @OneToMany(mappedBy = "vehicle")
    private List<Distribution> vehicleDistributionList;


    public Vehicle(Integer id){
        this.id = id;
    }

    public Vehicle(Integer id, String photo){
        this.id = id;
        this.photo = photo;
    }

}